include/indexstore/IndexStoreCXX.h

```diff
@@ -117,10 +141,22 @@ class IndexStore {
   friend class IndexUnitReader;
 
 public:
-  IndexStore(StringRef path, IndexStoreLibraryRef library, std::string &error) : library(std::move(library)) {
+  IndexStore(StringRef path, IndexStoreLibraryRef library,
+             IndexStoreCreationOptions *options,
+             std::string &error) : library(std::move(library)) {
     llvm::SmallString<64> buf = path;
     indexstore_error_t c_err = nullptr;
-    obj = api().store_create(buf.c_str(), &c_err);
+    // Backwards compatibility for previous versions which don't support a
+    // prefix mapping.
+    //
+    // TODO: Should we error if the API is unavailable but a mapping is given?
```

Comment:
Just that the mapping wouldn't actually do anything, which seems odd to me.

Reply:
Yeah I think failing is probably better, so I'll add that if you think it's alright?

Reply:
I had the feeling @benlangmuir was disagreeing with me there... I assume the idea being that you'd still want to be able to use the old API without having to change the options you pass? So in that case it's better to silently ignore them than hard error.

Reply:
I'm not against an error, just making sure it make sense.  If this were a command-line tool I'd say this should be a warning.  If we wanted to be fully flexible we could have a specific error type/enum-case at the Swift level, which would allow the code that uses IndexStoreDB to decide how to handle it (e.g. emit a warning and drop the mapping option, or keep it a hard error).  On the other hand, unless someone is planning to use that flexibility maybe it's not worth it and we could just make it a normal error with a string :shrug: Errors are always an open type, so we always have the option to add more specific errors in the future without breaking source.

Reply:
I was thinking an error makes sense since if you give path remappings it presumably mean the index data uses them, thus if we don't apply them you'll get weird paths.

Reply:
Added an error